telemetry/definitions/commonDefinitions.json

```diff
@@ -3655,6 +3655,32 @@
                 }
             ]
         },
+        {
+            "name": "codeTransform_projectDetails",
+            "description": "Transform initiates project validation before user interaction and we want to gather project details then.",
+            "metadata" : [
+                {
+                    "type": "codeTransformSessionId",
+                    "required": true
+                },
+                {
```

Comment:
Result can be a failure if the project is not valid for transform in which case the optional reason can be filled with more details as well.